examples/rust/opensk/src/clock.rs

```diff
@@ -13,18 +13,43 @@
 // limitations under the License.use opensk_lib::api::clock::Clock;
 
 use opensk_lib::api::clock::Clock;
+use wasefire::clock::Mode::Oneshot;
+use wasefire::clock::{Handler, Timer};
 
 #[derive(Default)]
-pub struct WasefireClock {}
+struct ClockHandler;
+
+impl Handler for ClockHandler {
+    fn event(&self) {
+        // TODO: What should we do here?
+        wasefire::debug!("timer elapsed");
+    }
+}
+
+pub struct WasefireTimer {
+    timer: Timer<ClockHandler>,
+}
+
+impl Default for WasefireTimer {
```

Comment:
Let's not implement `Default`. We can only create a timer with `make_timer`.

Reply:
This is required from the `Clock` trait: https://github.com/google/OpenSK/blob/5fdc6e0739b1a9e758d99fa0dc268282e7bc8aae/libraries/opensk/src/api/clock.rs#L22

Reply:
Oh thanks, I wasn't aware of that.